src/plots/cartesian/axes.js

```diff
@@ -84,28 +84,72 @@ function expandRange(range) {
  *     extraOption if there is no axis)
  * extraOption: aside from existing axes with this letter, what non-axis value is allowed?
  *     Only required if it's different from `dflt`
+ * domainRef: true if ' domain' should be appended to the axis items in the list
+ *     of possible values for this axis reference.
  */
-axes.coerceRef = function(containerIn, containerOut, gd, attr, dflt, extraOption) {
+axes.coerceRef = function(containerIn, containerOut, gd, attr, dflt, extraOption, domainRef) {
     var axLetter = attr.charAt(attr.length - 1);
     var axlist = gd._fullLayout._subplots[axLetter + 'axis'];
     var refAttr = attr + 'ref';
     var attrDef = {};
 
-    if(!dflt) dflt = axlist[0] || extraOption;
+    if(!dflt) dflt = axlist[0] || (typeof extraOption === 'string' ? extraOption : extraOption[0]);
     if(!extraOption) extraOption = dflt;
+    if(domainRef) axlist = axlist.concat(axlist.map(function(x) { return x + ' domain'; }));
```

Comment:
`concat`ing an item to the end of an array could be slower than using `push`.
You may rewrite 
```js
axlist = axlist.concat(axlist.map(function(x) { return x + ' domain'; }));
```
as
```js
axlist.push(axlist.map(function(x) { return x + ' domain'; }));
```